### PR TITLE
doc: adding reviewer guidelines

### PR DESCRIPTION
**Pull request by asettle**
Signed-off-by: Alexandra Settle <asettle@suse.com>

### CODE REVIEW

**Review by jhesketh — DISMISSED** [review]
Great start, thanks :-).

I think we should separate out `reviewing pull requests` and `merging pull requests`.

`reviewing pull requests` should cover what to check for (it's probably self explanatory so can likely be short).

`merging pull requests` should cover what is required before a PR is merged, but also when to merge a PR.
For example: All CI has run and is passing, at least 1 positive review/approval, open for feedback for at least 24 hours, no outstanding requests for changes, if still open after $days okay to self-merge? etc.
This should also clarify the needs-rebase case hopefully.

(just my 2cents anyway)

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -30,6 +30,26 @@ We use GitHub templates to manage the requests. You will either select:
 
 Ensure the appropriate label is selected before submitting your issue.
 
+## Reviewing Pull Requests
+
+Each pull request requires at least *one* review and approval from an active
+contributor. The primary function of reviews is to ensure code contributors
+are not blocked.
```
Comment (ml8mr) [inline CONTRIBUTING.md:37]: IMO, the primary function of reviews is to ensure quality. However, we also want PRs to be reviewed quickly, to avoid the blocking problem.

Comment (jecluis) [reply CONTRIBUTING.md:37]: Agree with @ml8mr

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -30,6 +30,26 @@ We use GitHub templates to manage the requests. You will either select:
...
+contributions. 
+
+All pull requests should be reviewed, even if the request is
+tagged with the label `needs-rebase`. At this time, it is imperative that
+code contributions move swiftly, and without minor blockers.
+
+Note: If the pull request requires a rebase, it is suitable to approve the
+pull request, the pull request will not merge automatically.
+
+The only pull requests that do not require a review is anything in draft status,
```
Comment (jecluis) [inline CONTRIBUTING.md:49]: I'd argue that even draft PRs should be reviewed if a review is requested.

IMO, if a PR is opened is because the author wants/needs feedback, even if it's a draft. Same goes for WIP.

**Review by jecluis — COMMENTED** [review]
I don't agree with not reviewing PRs based on draft or WIP. A PR with a request for review should always be reviewed. The whole point of drafts or WIP (IMO) is to make it clear the work is not final, but that should not discourage (heck, it should encourage!) reviews.

File: CONTRIBUTING.md (hunk 2 above)
Comment (l-mb) [reply CONTRIBUTING.md:49]: The keyword is "require"; draft PRs don't need one (but can request feedback). But then, draft/WIP won't get merged, and will require a mandatory review prior to merge, hence this may be redundant. Maybe rephrase paragraph to:

```suggestion
If your pull request is not yet in the state where it can be meaningfully reviewed and merged, please leave it in draft status or tag with the `WIP` label. You can of course still ask for feedback by manually assigning a reviewer or pinging them directly at this stage.
```

**Review by l-mb — DISMISSED** [review]
LGTM, possibly with the one paragraph altered as discussed (but not a blocking requirement from my side).

**Review by l-mb — APPROVED** [review]
LGTM.

**Review by jecluis — APPROVED** [review]

**Review by s0nea — DISMISSED** [review]

**Review by ml8mr — APPROVED** [review]

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -30,6 +30,48 @@ We use GitHub templates to manage the requests. You will either select:
...
+
+Before merging, check the following:
+
+- Ensure all CI has run and is passing
+- The pull request has been open for 24 hours for an appropriate feedback loop time
+- At least one reviewer has approved
+- There is no outstanding change requests
+
+If all of these are complete, you can self-merge (or merge on someone's behalf,
+if you have communicated your intention to do so).
```
Comment (jhesketh) [inline CONTRIBUTING.md:73]: Can you please clarify if this means we should not merge a PR unless we've discussed it with the author. And otherwise it is up to the author to merge their own pull request after approvals?

**Review by l-mb — APPROVED** [review]
Let's get this merged.

**Review by s0nea — APPROVED** [review]
